File: Backend/src/models/budget-model.ts
import { Budget } from "@prisma/client"

export interface CreateBudgetRequest {
    game_id: number
    allocated_budget: number
    allocated_currency: number
    allocated_tickets: number
    remaining_budget: number
    remaining_currency: number
    remaining_tickets: number
}

export interface BudgetResponse {
    budget_id: number
    game_id: number
    allocated_budget: number
    allocated_currency: number
    allocated_tickets: number
    remaining_budget: number
    remaining_currency: number
    remaining_tickets: number
}

export function toBudgetResponse(budget: Budget): BudgetResponse {
    return {
        budget_id: budget.budget_id,
        game_id: budget.game_id,
        allocated_budget: budget.allocated_budget,
        allocated_currency: budget.allocated_currency,
        allocated_tickets: budget.allocated_tickets,
        remaining_budget: budget.remaining_budget,
        remaining_currency: budget.remaining_currency,
        remaining_tickets: budget.remaining_tickets
    }
}

File: Frontend/app/src/main/java/com/visprog/starchive/models/BudgetModel.kt
package com.visprog.starchive.models

import com.google.gson.annotations.SerializedName

data class GetAllBudgetResponse(
    val data: List<BudgetModel>
)

data class GetBudgetResponse(
    val data: BudgetModel
)

data class BudgetModel(
    @SerializedName("budget_id") val budgetId: Int,
    @SerializedName("game_id") val gameId: Int,
    val allocated_budget: Float,
    val allocated_currency: Float,
    val allocated_tickets: Float,
    val remaining_budget: Float,
    val remaining_currency: Float,
    val remaining_tickets: Float
)

data class BudgetRequest(
    val gameId: Int,
    val allocated_budget: Float,
    val allocated_currency: Float,
    val allocated_tickets: Float,
    val remaining_budget: Float,
    val remaining_currency: Float,
    val remaining_tickets: Float
)